src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/policy/PolicyCommonBuilder.java

```diff
@@ -0,0 +1,204 @@
+
+            try {
+                ComponentItem.Builder builder = new ComponentItem.Builder()
+                                                    .applyCategory(MessageBuilderConstants.CATEGORY_TYPE_POLICY)
+                                                    .applyOperation(operation)
+                                                    .applyPriority(priority)
+                                                    .applyCategoryItem(policyNameItem)
+                                                    .applyCategoryGroupingAttribute(nullablePolicySeverityItem)
+                                                    .applyAllComponentAttributes(policyAttributes)
+                                                    .applyNotificationId(notificationId);
+                componentBuilderUtil.applyComponentInformation(builder, blackDuckResponseCache, componentData);
+                componentItems.add(builder.build());
+            } catch (Exception ex) {
+                logger.info("Error building policy component for notification {}, operation {}, component {}, component version {}", notificationId, operation, componentData.getComponentName(), componentData.getComponentVersionName());
+                logger.error("Error building policy component cause ", ex);
+            }
+        }
+        return componentItems;
+    }
+
+    public SetMap<ComponentVersionStatus, PolicyInfo> createComponentToPolicyMapping(
+        Collection<ComponentVersionStatus> componentVersionStatuses, Map<String, PolicyInfo> policyItems) {
+        SetMap<ComponentVersionStatus, PolicyInfo> componentToPolicyMapping = SetMap.createDefault();
+        for (ComponentVersionStatus componentVersionStatus : componentVersionStatuses) {
+            Set<PolicyInfo> componentPolicies = getPoliciesForComponent(componentVersionStatus, policyItems);
+            componentToPolicyMapping.addAll(componentVersionStatus, componentPolicies);
+        }
+        return componentToPolicyMapping;
+    }
+
+    public Set<PolicyInfo> getPoliciesForComponent(ComponentVersionStatus componentVersionStatus, Map<String, PolicyInfo> policyItems) {
+        return componentVersionStatus.getPolicies().stream()
+                   .filter(policyItems::containsKey)
+                   .map(policyItems::get)
+                   .collect(Collectors.toSet());
+    }
+
+    public boolean hasVulnerabilityRule(VersionBomPolicyRuleView policyRule) {
+        PolicyRuleExpressionSetView expression = policyRule.getExpression();
+        return hasVulnerabilityRule(expression);
+    }
+
+    public boolean hasVulnerabilityRule(PolicyRuleView policyRule) {
+        PolicyRuleExpressionSetView expression = policyRule.getExpression();
+        return hasVulnerabilityRule(expression);
+    }
+
+    private boolean hasVulnerabilityRule(PolicyRuleExpressionSetView expressionSet) {
+        List<PolicyRuleExpressionView> expressions = expressionSet.getExpressions();
```

Comment:
You can change this method to public, remove the two methods above and have the user instead pass a List<PolicyRuleExpressionView> to the method. Will help focus on passing the the most basic item possible.